etcdserver/server.go

```diff
@@ -146,16 +146,20 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 	var n raft.Node
 	var s *raft.MemoryStorage
 	var id types.ID
-	walVersion, err := wal.DetectVersion(cfg.DataDir)
+	var ss *snap.Snapshotter
+	baseDataDir := cfg.DataDir
+	// Data dir lives under member now.
+	cfg.DataDir = path.Join(cfg.DataDir, "member")
```

Comment:
I don't like this way.
We have function cfg.SnapDir(), which relies on the value of cfg.DataDir. And it cannot behave correctly until you set the value here. It is uncontrollable.